src/test/rustdoc/trait-self-link.rs

```diff
@@ -1,6 +0,0 @@
-// @!has trait_self_link/trait.Foo.html //a/@href ../trait_self_link/trait.Foo.html
```

Comment:
Instead of deleting this test, it might be better to just change this from `@!has` to `@has` so that we know when the behavior changes in the future.